app/controllers/users_controller.rb

```diff
@@ -168,17 +168,23 @@ def update
 
   def changed_email
     if !params[:new_email].blank? && reauthenticate
-      @old_email = @user.email
-      @user.email = params[:new_email]
       @new_email = params[:new_email]
-      @confirm_email = params[:email_confirmation]
 
-      if @new_email == @confirm_email && @user.save
-        flash[:notice] = ts('Your email has been successfully updated')
+      if @new_email != params[:email_confirmation]
+        flash[:error] = ts("Email addresses don't match! Please retype and try again")
+        return
+      end
+
+      @old_email = @user.email
```

Comment:
I think `@old_email` and `@new_email` can both be local variables instead of instance variables. I don't see them [used](https://stackoverflow.com/questions/18855178/how-are-rails-instance-variables-passed-to-views) in the [view](https://github.com/otwcode/otwarchive/blob/63ed5aa8387b7593831811e66a2f2c2654bdea15/app/views/users/change_email.html.erb).